worlds/tunic/__init__.py

```diff
@@ -412,12 +429,34 @@ def fill_slot_data(self) -> Dict[str, Any]:
             "disable_local_spoiler": int(self.settings.disable_local_spoiler or self.multiworld.is_race),
         }
 
+        # this would be in a stage if there was an appropriate stage for it
+        self.player_item_link_locations = {}
+        groups = self.multiworld.get_player_groups(self.player)
+        if groups:
+            if not self.item_link_locations:
+                tunic_worlds: Tuple[TunicWorld] = self.multiworld.get_game_worlds("TUNIC")
+                # figure out our groups and the items in them
+                for tunic in tunic_worlds:
+                    for group in self.multiworld.get_player_groups(tunic.player):
+                        self.item_link_locations.setdefault(group, {})
+                for location in self.multiworld.get_locations():
+                    if location.item.player in self.item_link_locations.keys():
```

Comment:
This would currently raise an error if there is a location in the multiworld without an item. There is some debate about whether this should be fixed or if a world supplying more locations than items should be considered at fault instead (various AP worlds fail when there are unfilled locations and there's a PR to make unfilled locations crash generation).

```suggestion
                    if location.item and location.item.player in self.item_link_locations.keys():
```